contracts/utils/CAIP2.sol

```diff
@@ -12,6 +12,11 @@ import {Strings} from "./Strings.sol";
  * chain_id:    namespace + ":" + reference
  * namespace:   [-a-z0-9]{3,8}
  * reference:   [-_a-zA-Z0-9]{1,32}
+ *
+ * WARNING: The CAIP-2 identifier does not require a canonical representation. In the EVM context, it is
```

Comment:
I'm not sure what "The CAIP-2 identifier does not require a canonical representation" means.

Can you elaborate ?

Reply:
The point is that CAIP2 does not define a "canonical" representation for each chain. Although EVM chains will use `erc155` as namespace, there may be chains that have more than one representation for a single chain.

Perhaps saying that "The CAIP-2 identifier may not be canonical" instead of "does not require a canonical representation".

Reply:
What about

> In some cases, multiple CAIP-2 identifiers may all be valid representation of a single chain. For EVM chains, it is recommended to use "eip155:xxx` with `xxx` the EIP-155 chain id as the canonical representation.

Reply:
I like it